test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -7112,6 +7112,44 @@ BOOST_AUTO_TEST_CASE(array_length_validation)
 	CHECK_ERROR(text, TypeError, "Invalid array length, expected integer literal.");
 }
 
+BOOST_AUTO_TEST_CASE(array_length_not_convertible_to_integer)
+{
+	char const* text = R"(
+		contract C {
+			uint[true] ids;
+		}
+	)";
+	CHECK_ERROR(text, TypeError, "Invalid array length, expected integer literal.");
+}
+
+BOOST_AUTO_TEST_CASE(array_length_invalid_expression)
+{
+	char const* text = R"(
+		contract C {
+			uint[-true] ids;
+		}
+	)";
+	CHECK_ERROR(text, TypeError, "Invalid constant expression.");
+	text = R"(
+		contract C {
+			uint[true/1] ids;
```

Comment:
Do we have a test for a division by zero or something like `2**900` in the array length description?

Reply:
`2**900` is tested for one test above these.

Division by zero will be addressed in the next PR by @wadeAlexC.

Reply:
ok!